components/alertbanner/stories/template.js

```diff
@@ -53,29 +54,122 @@ export const Template = ({
 						})}>${text}</p>
 					`)}
 				</div>
-				${when(hasActionButton, () =>
+				${when(actionButtonText, () =>
 					Button({
 						size: "m",
 						staticColor: "white",
 						treatment: "outline",
-						label: "Action",
+						label: actionButtonText,
 					}, context)
 				)}
 			</div>
-			<div class=${classMap({
-				[`${rootClass}-end`]: true
-			})}>
-				${Divider({
-					vertical: true,
-					size: "s",
-					tag: "div",
-				}, context)}
-				${CloseButton({
-					size: "m",
-					staticColor: "white",
-					onclick,
-				}, context)}
-			</div>
+			${when(showCloseButton, () => html`
+				<div class=${classMap({
+					[`${rootClass}-end`]: true
+				})}>
+					${Divider({
+						vertical: true,
+						size: "s",
+						tag: "div",
+					}, context)}
+					${CloseButton({
+						size: "m",
+						staticColor: "white",
+						onclick,
+					}, context)}
+				</div>
+			`)}
 		</div>
 	`;
 };
+
+/**
+ * Examples of all text overflow (wrapping) scenarios.
+ */
+export const TextOverflowTemplate = (args) => html`
+	<div
+		style=${styleMap({
+			display: "flex",
+			flexDirection: "column",
+			gap: "12px",
+		})}
+	>
+		${Template({
+			...args,
+			variant: "info",
+			text: "Your trial will expire in 3 days. Once it expires your files will be saved and ready for you to open again once you have purchased the software.",
+		})}
+		${Template({
+			...args,
+			variant: "neutral",
+			text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
+			actionButtonText: "",
+		})}
+	</div>
+`;
+
+/**
+ * Examples of options for contextual action button and close button.
+ */
+export const ActionableOptionsTemplate = (args, context) => Container({
```

Comment:
This is overall so well organized and makes it so clear, but this especially is wonderful! Paired with the JSDoc comment that comes before it in alertbanner.stories.js, it makes it so easy to understand the different combinations of buttons and when/how to use them. 🏆